Source/Dafny/Resolver.cs

```diff
@@ -4423,7 +4423,7 @@ public class ModuleBindings {
         Contract.Assert(cl.TypeArgs.Count == B.Count);
         moreXConstraints = false;
         for (int i = 0; i < B.Count; i++) {
-          var msgFormat = "variance for type parameter" + (B.Count == 1 ? "" : "" + i) + " expects {0} {1} {2}";
+          var msgFormat = "variance for type parameter" + (B.Count == 1 ? "" : " " + i) + " expects {0} {1} {2}";
```

Comment:
Since the integer is an index and not an ordinal, would it be possible to say "at index 0" instead of just "0" ? It will help us be consistent with the error messages [after this recent change](https://github.com/dafny-lang/dafny/blob/master/Source/Dafny/Resolver.cs#L16993).
That way, it'll be less confusing when encountering two type parameters, the second one having an error,, and the error would currently say `variance for type parameter 1 expects ....`

I would even prefer to have ordinals here ( i + 1 ) so we wouldn't need the extra "at index".

Reply:
I'll change it to "at index 0" to be consistent with the other recent change.

(Btw, ordinals contain the natural numbers, and both start with 0. And I find it confusing to add arbitrary offsets, like 1. :))

Reply:
Interesting, I never thought of the ambiguity between "Ordinal" as a mathematical meaning (the natural numbers including zero + infinity + (infinity + 1) + ... and so on), and the English meaning of "ordinal" which is always 1-based (the first, the second, the third...) as opposed to "cardinal" which is the number itself (and can contain zero).
Prehaps we should simply have a converter from number to plain text English ordinals at some point.